File: pruebas.py
import requests

# Configura tus credenciales
client_id = "oauth-mkplace-oauthmmtivtzpbtrvgytvrupropro"
client_secret = "BPy6S-z@WPfKm~Q~"
token_url = "https://auth.inditex.com:443/openam/oauth2/itxid/itxidmp/access_token"

headers = {
    "User-Agent": "OpenPlatform/1.0"
}

data = {
    "grant_type": "client_credentials",
    "scope": "technology.catalog.read"
}

# Solicitar el token de acceso
try:
    response = requests.post(
        token_url,
        headers=headers,
        data=data,
        auth=(client_id, client_secret)
    )

    if response.ok:
        id_token = response.json().get("id_token")
        access_token = response.json().get("access_token")
    else:
        print("Error al obtener el token de acceso:", response.status_code, response.text)
        exit()

except requests.exceptions.RequestException as e:
    print(f"Error en la solicitud: {e}")
    exit()

# Realiza la solicitud de búsqueda de productos
url = "https://api.inditex.com/pubvsearch/products"
headers = {
    'Authorization': f'Bearer {id_token}',  # Asegúrate de incluir "Bearer" antes del token
    'Content-Type': 'application/json',
    'User-Agent': 'OpenPlatform/1.0'
}

params = {
    'image': 'https://shop.rfef.es/cdn/shop/products/23CM0743_Z4.jpg',
    'page': 1,
    'perPage': 50
}

try:
    response = requests.get(url, headers=headers, params=params)

    if response.ok:
        print(response.json())  # Muestra la respuesta en formato JSON
    else:
        print("Error al hacer la solicitud de búsqueda de productos:", response.status_code, response.text)

except requests.exceptions.RequestException as e:
    print(f"Error en la solicitud: {e}")
    exit()

File: Front_Hack/pruebas.js
import {clientId ,clientSecret,tokenUrl} from './credentials.mjs';
// Configure your credentials


// Function to encode credentials for Basic Auth
function getBasicAuthHeader(clientId, clientSecret) {
  return 'Basic ' + Buffer.from(`${clientId}:${clientSecret}`).toString('base64');
}

// Function to get access token
async function getAccessToken() {
  try {
    const headers = {
      'User-Agent': 'OpenPlatform/1.0',
      'Content-Type': 'application/x-www-form-urlencoded',
      'Authorization': getBasicAuthHeader(clientId, clientSecret)
    };

    const formData = new URLSearchParams();
    formData.append('grant_type', 'client_credentials');
    formData.append('scope', 'technology.catalog.read');

    const response = await fetch(tokenUrl, {
      method: 'POST',
      headers: headers,
      body: formData
    });

    if (!response.ok) {
      throw new Error(`Error getting access token: ${response.status} ${await response.text()}`);
    }

    const data = await response.json();
    return {
      idToken: data.id_token,
      accessToken: data.access_token
    };
  } catch (error) {
    console.error(`Request error: ${error.message}`);
    throw error;
  }
}

// Function to search products
async function searchProducts(idToken) {
  try {
    const url = "https://api.inditex.com/pubvsearch/products";
    const headers = {
      'Authorization': `Bearer ${idToken}`,
      'Content-Type': 'application/json',
      'User-Agent': 'OpenPlatform/1.0'
    };

    // Create URL with parameters
    const params = new URLSearchParams({
      'image': 'https://shop.rfef.es/cdn/shop/products/23CM0743_Z4.jpg',
      'page': 1,
      'perPage': 50
    });

    const response = await fetch(`${url}?${params}`, {
      method: 'GET',
      headers: headers
    });

    if (!response.ok) {
      throw new Error(`Error searching products: ${response.status} ${await response.text()}`);
    }

    return await response.json();
  } catch (error) {
    console.error(`Request error: ${error.message}`);
    throw error;
  }
}

// Main execution function
async function main() {
  try {
    const tokens = await getAccessToken();
    console.log("Access token obtained successfully");
    
    const searchResults = await searchProducts(tokens.idToken);
    console.log(JSON.stringify(searchResults, null, 2));
  } catch (error) {
    console.error("Execution failed:", error);
    process.exit(1);
  }
}

// Execute the script
main();